cluster/config_info_retriever.go

```diff
@@ -9,9 +9,9 @@ type ConfigInfoRetriever struct {
 }
 
 // NewConfigInfoRetriever returns a *ConfigInfoRetriever
-func NewConfigInfoRetriever(c *config.Config) *ConfigInfoRetriever {
+func NewConfigInfoRetriever(config config.ConfigInfoRetrieverConfig) *ConfigInfoRetriever {
```

Comment:
since the package is already called `config`, can't you remove the `Config` prefix from the struct name?